uaclient/apt.py

```diff
@@ -72,6 +83,12 @@
 event = event_logger.get_event_logger()
 LOG = logging.getLogger(util.replace_top_level_logger_name(__name__))
 
+# Most part of the apt_pkg functionality needs the module to be initialized.
+# This call is checking for the 'Dir' configuration - which needs to be there
+# for apt_pkg to be ready - and if it is empty we initialize.
+if apt_pkg.config.get("Dir") == "":
```

Comment:
Well that works for uaclient, but how about another module imported something from us and already has apt_pkg initialized with some change? 
This is, yes, internal logic from apt_pkg, but it is a) a basic configuration that does indicate initialization and b) a mandatory config not likely to ever change.

About the decorator, or any other method of checking/initializing when the function is called, that is fair, but I would still check for the 'Dir' config at call time and it would make no difference from initializing on import IMHO.